numpy/f2py/src/fortranobject.c

```diff
@@ -130,8 +130,7 @@ format_def(char *buf, Py_ssize_t size, FortranDataDef def)
         return -1;
     }
 
-    p[size] = ')';
-    p++;
+    *p++ = ')';
```

Comment:
`size` is decremented on the next line.

Throughout the function, the logic seems to be: `p` is the current pointer within the buffer, `size` is the amount of bytes left to write in the buffer.